src/NHibernate/Impl/ExpressionProcessor.cs

```diff
@@ -105,7 +106,9 @@ public string AsProperty()
 		private readonly static IDictionary<LambdaSubqueryType, IDictionary<ExpressionType, Func<string, DetachedCriteria, AbstractCriterion>>> _subqueryExpressionCreatorTypes;
 		private readonly static IDictionary<string, Func<MethodCallExpression, ICriterion>> _customMethodCallProcessors;
 		private readonly static IDictionary<string, Func<Expression, IProjection>> _customProjectionProcessors;
-
+		private static readonly Dictionary<ExpressionType, ISQLFunction> _binaryArithmethicTemplates = new Dictionary<ExpressionType, ISQLFunction>();
```

Comment:
Use same pattern in the class:
```suggestion
		private readonly static IDictionary<ExpressionType, ISQLFunction> _binaryArithmethicTemplates = new Dictionary<ExpressionType, ISQLFunction>();
```

Reply:
@lillo42 it's not practical to use a collection interface for the fields. Using the class allows you to avoid boxing and use some optimizations (no-boxing iterators, for example).

Reply:
Ok, I understand

(off-topic) Should we change the ```IDictionary<T>```  to ```Dictionary<T>``` ?